docs/httpparse-error-handling.md

```diff
@@ -1,16 +1,36 @@
 # *Flexirest:* HTTP/parse error handling
 
-Sometimes the backend server may respond with a non-200/304 header, in which case the code will raise an `Flexirest::HTTPClientException` for 4xx errors or an `Flexirest::HTTPServerException` for 5xx errors. These both have a `status` accessor and a `result` accessor (for getting access to the parsed body):
+## HTTP Errors
+
+Sometimes the backend server may respond with a non-2xx/3xx header, in which case the code will raise an `Flexirest::HTTPClientException` for 4xx errors or an `Flexirest::HTTPServerException` for 5xx errors. 
+
+These both have a `status` accessor and a `result` accessor (for getting access to the parsed body). However, you can make use of the `message` method (or the `to_s` method) that will build a nice error message for you (containing the URL and the HTTP method used for the request as well as the HTTP status code and the original body of the response).
 
 ```ruby
 begin
   Person.all
 rescue Flexirest::HTTPClientException, Flexirest::HTTPServerException => e
-  Rails.logger.error("API returned #{e.status} : #{e.result.message}")
+  # Display the HTTP status and parsed body
+  puts "Backend returned HTTP #{e.status} with following parsed body: #{e.result}"
+  
+  # Using the message method, same than calling the to_s method
+  puts e.message
 end
 ```
 
-If the response is unparsable (e.g. not in the desired content type), then it will raise an `Flexirest::ResponseParseException` which has a `status` accessor for the HTTP status code and a `body` accessor for the unparsed response body.
+If you still wish to get access to the original body response from the backend apart from the `message` method, you can do it by accessing its `raw_response` instance variable:
+
+```ruby
+begin
+  Person.all
+rescue Flexirest::HTTPClientException, Flexirest::HTTPServerException => e
+  puts "Original body is: #{e.instance_variable_get(:@raw_response)}"
+end
+```
+
+## Parse Error
+
+If the original response is unparsable (e.g. not in the desired content type), then it will raise an `Flexirest::ResponseParseException` which has a `status` accessor for the HTTP status code and a `body` accessor for the unparsed response body.
```

Comment:
Is the `body` not the same as `e.instance_variable_get(:@raw_response)`?

Reply:
Yes but there are two Exception classes :
- ResponseParseException which uses a body attribute
- HTTPException which uses the raw_response attribute

However I think that both handle the same @response.body value unparsed.
An alignment should be possible.

Reply:
I don't have any opinion which is the better official new name going forward, but alias the old one to the new one. You OK adding this to your PR?

Reply:
Yes ok, I have added a commit.

Reply:
Love that commit, happy to accept that.